gammapy/background/tests/test_energy_offset_array.py

```diff
@@ -1,21 +1,80 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 from __future__ import absolute_import, division, print_function, unicode_literals
+from numpy.testing import assert_equal
 import numpy as np
-from ..energy_offset_array import EnergyOffsetArray
+from astropy.table import Table
 from ...data import DataStore
 from ...datasets import gammapy_extra
 from ...utils.testing import requires_dependency, requires_data
 from ...utils.energy import EnergyBounds
+from ..energy_offset_array import EnergyOffsetArray
+from ...data import EventList
+
+
+def make_test_array():
+    ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 100, 'TeV')
+    offset = np.linspace(0, 2.5, 100)
+    array = EnergyOffsetArray(ebounds, offset)
+    # put some dummy example data
+    # TODO: this could be made optional or split out into a
+    # separate utility function later on when lookup / interpolation methods are added to the class.
+    # array.data[5, 3] = 42
+    return array
 
 
 @requires_dependency('matplotlib')
 @requires_data('gammapy-extra')
-def test_energy_offset_array():
+def test_energy_offset_array_fill():
     dir = str(gammapy_extra.dir) + '/datasets/hess-crab4'
     data_store = DataStore.from_dir(dir)
     ev_list = data_store.load_all('events')
-    ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 100, 'TeV')
-    offset = np.linspace(0, 2.5, 100)
-    array = EnergyOffsetArray(ebounds, offset)
+
+    array = make_test_array()
     array.fill_events(ev_list)
+
+    # TODO: add some assert, e.g. counts in some bin with non-zero entries.
+
+
+@requires_dependency('scipy')
+def test_energy_offset_array_fill_evaluate():
+    array = make_test_array()
+    # Define an EventList with three events
```

Comment:
Move this code that fills the events to the `make_test_array` function at the top.
A test array that's not all zeros will be useful for other tests, e.g. smooth or radial acceptance curve extract or ...